sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/dofn/DetectNewPartitionsDoFn.java

```diff
@@ -109,9 +110,7 @@ public Instant getInitialWatermarkEstimatorState(@Element PartitionMetadata part
   public TimestampRange initialRestriction(@Element PartitionMetadata partition) {
     final com.google.cloud.Timestamp createdAt = partition.getCreatedAt();
     return TimestampRange.of(
-        com.google.cloud.Timestamp.ofTimeSecondsAndNanos(
-            createdAt.getSeconds(), createdAt.getNanos() - 1),
-        com.google.cloud.Timestamp.MAX_VALUE);
+        TimestampUtils.previous(createdAt), com.google.cloud.Timestamp.MAX_VALUE);
```

Comment:
Makes sure there is no underflow here by using `TimestampUtils.previous` instead of simply subtracting `1ns`